templates/app/helpers/spree/base_helper.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require 'carmen'
+
+module Spree
+  module BaseHelper
```

Comment:
I copied `Spree::BaseHelper` from Solidus Core in order to override some methods in the helper that call spree routes e.g. the `logo` helper which originally called `spree.root_path`. I wasn't able to figure out a way to override only those helpers used by the frontend. I tried the following:

1. Overriding `logo` in ApplicationHelper.
2. Overriding `logo` in a new local BaseHelper and setting `helper 'base'` in StoreController.

It appears the `Spree::Base` helper from the Solidus engine would always have higher precedence over local helpers from the app.

When copying `Spree::BaseHelper`, I had to override all the methods because once the `Spree::BaseHelper` module is overridden locally, it seems the app would no be longer to find non-overridden methods from the `Spree::BaseHelper` in the Solidus gem.

The only alternative I can think of is to replace the `Spree::Base` helpers in with local variations, something like:

```
module ApplicationHelper
    def app_logo(image_path = Spree::Config[:logo])
      link_to image_tag(image_path), root_path
    end
end
```

Let me know what you think!

Reply:
> However, the biggest downside with this approach is that we can only remove these helpers on a major release, since they are considered public API. Is that correct?

Yes, but we can deprecate and stop using them before the removal.

Reply:
@kennyadsl Just to clarify: if we just deprecate the helpers in `Spree::BaseHelpers` in Solidus Core, we'd still need to keep the local BaseHelpers module under the Spree namespace in order to allow them to override the Solidus Core helpers. It is only when we actually delete the Solidus Core helpers that we can finally remove local BaseHelpers module from the Spree namespace.